tiledb/sm/query/query.h

```diff
@@ -682,9 +682,25 @@ class Query {
    */
   Status set_subarray(const void* subarray);
 
+  /**
+   * Sets the query subarray.
+   *
+   * @param subarray The subarray to be set.
+   * @return Status
+   *
+   * @note Setting a subarray for sparse arrays, or for dense arrays
+   *     when performing unordered (sparse) writes, has no effect
+   *     (will be ingnored).
+   */
+  Status set_subarray(const tiledb::sm::Subarray* subarray);
+
   /** Sets the query subarray, without performing any checks. */
   Status set_subarray_unsafe(const NDRange& subarray);
 
+  /** Reference current Reader/Writer subarray according to query type */
+  const Subarray& subarray() const;
+  Subarray* subarray();
```

Comment:
separate doc for this routine. we rarely use a single block comment for more than one function.

Reply:
One of methods/prototypes now removed.